.github/workflows/compile-queries.yml

```
@@ -31,14 +29,17 @@ jobs:
           path: '*/ql/src/.cache'
           key: codeql-compile-pr-${{ github.sha }} # deliberately not using the `compile-compile-main` keys here.
           restore-keys: |
-            codeql-compile-main-${{ env.merge-base }}
-            codeql-compile-main-
+            codeql-compile-refs/heads/${{ github.base_ref }}-${{ env.merge-base }}
+            codeql-compile-refs/heads/${{ github.base_ref }}-
```

Perhaps fall back to `main` too, just in case. It won't matter much in practice as most PR will target `main` anyway.

```suggestion
            codeql-compile-refs/heads/${{ github.base_ref }}-
            codeql-compile-refs/heads/main-            
```